bin/src/hickory-dns.rs

```diff
@@ -472,13 +474,13 @@ fn run() -> Result<(), String> {
 
     if !args.disable_udp && !config.get_disable_udp() {
         // load all udp listeners
-        for udp_socket in &sockaddrs {
-            info!("binding UDP to {:?}", udp_socket);
-            let udp_socket = runtime
-                .block_on(UdpSocket::bind(udp_socket))
-                .map_err(|err| {
-                    format!("failed to bind to UDP socket address {udp_socket:?}: {err}")
-                })?;
+        for addr in &listen_addrs {
+            info!("binding UDP to {addr:?}");
+
+            let _guard = runtime.enter();
```

Comment:
Feels like it would make sense to just do this at the top level once instead of per protocol?